### PR TITLE
Generate normals

### PR DESCRIPTION
**Pull request by archanasri**
1. New shapeworks command: GenerateNormals
2. Consolidated commands: FixCuttingPlanes, GenerateFeatureGradientFiles, GenerateNormalFeaFiles

### CODE REVIEW

**Review by akenmorris — APPROVED** [review]
This all looks good to me.  I would probably recommend that we switch to PEP8 for our Python API though rather than come up with our own non-standard conventions.

File: Python/shapeworks/ShapeworksPython.cpp
Comment (medakk) [inline Python/shapeworks/ShapeworksPython.cpp:597]: Its not really clear what this means. "return (x,y,z) coordinates of the vertex at a given index" perhaps?

**Review by medakk — APPROVED** [review]

File: Python/shapeworks/ShapeworksPython.cpp
Comment (archanasri) [reply Python/shapeworks/ShapeworksPython.cpp]: Updated documentation

**Comment by akenmorris** [comment]
What is this generateNormals used for?  I noticed that splitting is on, so this may be problematic for some uses (but desired for others).  Also, I saw that we're also using vtkPolyDataNormals with splitting off in Mesh::fillHoles.

**Comment by cchriste** [comment]
Sorry, but I'm not 

> This all looks good to me. I would probably recommend that we switch to PEP8 for our Python API though rather than come up with our own non-standard conventions.

Can you please provide a specific example?

**Comment by cchriste** [comment]
> What is this generateNormals used for? I noticed that splitting is on, so this may be problematic for some uses (but desired for others). Also, I saw that we're also using vtkPolyDataNormals with splitting off in Mesh::fillHoles.

We discussed this during last week's meeting, and the two changes that were made after that were to use ->SplittingOff(); for the normal computation filter and call that function in Mesh::fillHoles() rather than using the filter directly (less cut and paste). These were both finished by Archana.

**Comment by cchriste** [comment]
The biggest concern I have is this pull request merges release_6.0c into master. Is that what we want to do? 
I thought 6.0 was all about meshes, which would reasonably include this function.

**Review by cchriste — COMMENTED** [review]
Wait until release is merged back to master, then merge this branch into master.

**Comment by akenmorris** [comment]
> Can you please provide a specific example?

Sure, a specific example would be something like:

"Function names should be lowercase, with words separated by underscores as necessary to improve readability."

https://www.python.org/dev/peps/pep-0008/#naming-conventions

This means that "generateNormals" should be "generate_normals".

More details are at #1149

**Pull request merged**
